### PR TITLE
upgrades: Add UI

### LINKED ISSUE #51378: upgrades: Update alternate UI to show migration progress, out-of-band migration errors, etc
Slap better content into the UI block from #51377.

cc @Strum355 - this could be a good area to start on simply extracting and formatting (in plain text) the current status of a migration (where did we come from, where are we going, how far are we, what migrations need to be applied, what oobmigrations need to be applied, any errors, etc). We'll end up slapping it in a UI somewhere, but updating the upgradestore (or wherever we end up implementing it) to understand the correct _progress_ and _state_ of the upgrade is going to be the fun part.

### PR DESCRIPTION
**Pull request by efritz**
Fixes #51378 
image
image

## Test plan

Tested locally.

edit: clabotthings

### CODE REVIEW

**Review by Numbers88s — APPROVED** [review]

File: cmd/frontend/internal/cli/autoupgrade.go [hunk 1]
```diff
@@ -198,20 +210,27 @@ func runMigration(ctx context.Context,
...
+
+	migratorsBySchema := map[string]dialer{
+		"frontend":     connections.MigrateNewFrontendDB,
+		"codeintel":    connections.MigrateNewCodeIntelDB,
+		"codeinsights": connections.MigrateNewCodeInsightsDB,
+	}
+	for schema, migrateLastMile := range migratorsBySchema {
+		obsvCtx.Logger.Info("Running last-mile migrations", log.String("schema", schema))
+
+		sqlDB, err := migrateLastMile(obsvCtx, dsns[schema], "frontend")
```
Comment (Strum355) [inline cmd/frontend/internal/cli/autoupgrade.go:231]:
```suggestion
		sqlDB, err := migrateLastMile(obsvCtx, dsns[schema], appName)
```
no big deal anymore (mostly just matters for the old connections block at the bottom) but just nice to have it named anyway

**Review by Strum355 — APPROVED** [review]
just the tiny thing and then LGTM :letsgooo:

**Comment by efritz** [comment]
@cla-bot check

**Comment by efritz** [comment]
@cla-bot check

**Comment by Strum355** [comment]
@cla-bot check or replaced by Cody

**Comment by davejrt** [comment]
@cla-bot check

**Comment by efritz** [comment]
@cla-bot check

**Comment by efritz** [comment]
@cla-bot check

**Comment by jhchabran** [comment]
@cla-bot check

**Comment by davejrt** [comment]
@cla-bot check

**Pull request merged**